--- etcdctl/BUILD.bazel	2018-12-13 18:36:55.835000000 -0500
+++ etcdctl/BUILD.bazel	2018-12-13 18:37:14.120000000 -0500
@@ -4,7 +4,7 @@
     name = "go_default_library",
     srcs = ["main.go"],
     importpath = "github.com/coreos/etcd/etcdctl",
-    visibility = ["//visibility:private"],
+    visibility = ["//visibility:public"],
     deps = [
         "//etcdctl/ctlv2:go_default_library",
         "//etcdctl/ctlv3:go_default_library",
