--- BUILD.bazel	2019-01-20 18:51:55.651000000 -0500
+++ BUILD.bazel	2019-01-20 18:52:18.021000000 -0500
@@ -4,7 +4,7 @@
     name = "go_default_library",
     srcs = ["main.go"],
     importpath = "k8s.io/kube-state-metrics",
-    visibility = ["//visibility:private"],
+    visibility = ["//visibility:public"],
     deps = [
         "//collectors:go_default_library",
         "//vendor/github.com/golang/glog:go_default_library",
