commit b0af0f68b102e37df474c4c21fa46935850f824b
Author: Jeff Grafton <jgrafton@google.com>
Date:   Thu Jun 7 15:30:39 2018 -0700

    gazelle: set proper importmap on dependencies in vendor/

diff --git a/hack/update-bazel.sh b/hack/update-bazel.sh
index 98def35f9f..10fb3b3800 100755
--- a/hack/update-bazel.sh
+++ b/hack/update-bazel.sh
@@ -36,6 +36,12 @@ go install ./vendor/github.com/bazelbuild/bazel-gazelle/cmd/gazelle
 go install ./vendor/github.com/kubernetes/repo-infra/kazel
 
 touch "${KUBE_ROOT}/vendor/BUILD"
+# Ensure that we use the correct importmap for all vendored dependencies.
+# Probably not necessary in gazelle 0.13+
+# (https://github.com/bazelbuild/bazel-gazelle/pull/207).
+if ! grep -q "# gazelle:importmap_prefix" "${KUBE_ROOT}/vendor/BUILD"; then
+  echo "# gazelle:importmap_prefix k8s.io/kubernetes/vendor" >> "${KUBE_ROOT}/vendor/BUILD"
+fi
 
 gazelle fix \
     -build_file_name=BUILD,BUILD.bazel \
diff --git a/vendor/BUILD b/vendor/BUILD
index 1c1e2e5713..09df8b1e01 100644
--- a/vendor/BUILD
+++ b/vendor/BUILD
@@ -1,3 +1,5 @@
+# gazelle:importmap_prefix k8s.io/kubernetes/vendor
+
 filegroup(
     name = "package-srcs",
     srcs = glob(["**"]),
