commit 4c7c058a61516ce11b381bb1e58fd20fd928e9be
Author: Jeff Grafton <jgrafton@google.com>
Date:   Fri May 18 15:21:26 2018 -0700

    Remove go_prefix from root BUILD file

diff --git a/build/root/BUILD.root b/build/root/BUILD.root
index 38e89c12e9..85212cb607 100644
--- a/build/root/BUILD.root
+++ b/build/root/BUILD.root
@@ -10,11 +10,8 @@
 
 package(default_visibility = ["//visibility:public"])
 
-load("@io_bazel_rules_go//go:def.bzl", "go_prefix")
 load("@io_kubernetes_build//defs:build.bzl", "gcs_upload")
 
-go_prefix("k8s.io/kubernetes")
-
 filegroup(
     name = "_binary-artifacts-and-hashes",
     srcs = [
